File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOneXMLValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OneXMLInValue.OneXMLValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOneXMLValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOneXMLValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.w3c.dom.Element.class,
                OneXMLValue.class.getDeclaredField("oneXML").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.w3c.dom.Element.class,
                OneXMLValue.class.getMethod("getOneXML").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OneXMLValue.class,
                OneXMLValue.class.getMethod("setOneXML", org.w3c.dom.Element.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OneXMLValue().getOneXML());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOneXMLValue.scala
package com.dslplatform.ocd.values

import OneXMLInValue.OneXMLValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOneXMLValue extends SpecExtensions {
  def `Test property field type: scala.xml.Elem`() =
    checkType(
      typeOf[scala.xml.Elem]
    , typeOf[OneXMLValue].member("oneXML": TermName).asMethod.returnType
    )

  def `Test property default value: null`() {
    assert(OneXMLValue().oneXML === null)
  }
}
